packages/react-dom/src/__tests__/ReactDeprecationWarnings-test.internal.js

```diff
@@ -69,4 +69,50 @@ describe('ReactDeprecationWarnings', () => {
         '\n    in Component (at **)',
     );
   });
+
+  it('should not warn when owner and self are the same for string refs', () => {
```

Comment:
This says _not_ but the test says it warns. Probably because it's concurrent mode which implies strict.

Try using `ReactNoop.renderLegacySyncRoot` instead.